### ISSUE TITLE
Bout title in email notification differs depending on who posted

### ISSUE THREAD

**Issue by carlosmiranda** [yegor256/netbout#566]
When I receive email notifications, I get different subject lines based on who sent the last message. As you can see below, the messages sent by me contain the subject line:

> #35023: Carlos Miranda

Meanwhile, the ones that were sent by other people (in this case, "yegor256" and "alice") are entitled:

> #35023: @carlosmiranda

![image](https://cloud.githubusercontent.com/assets/1917892/6317943/c2a4fade-baaf-11e4-9094-517244e42f18.png)

On Netbout itself, the title is set to "Carlos Miranda"

![image](https://cloud.githubusercontent.com/assets/1917892/6317955/6266848e-bab0-11e4-9355-9c4463432191.png)

Because of this, Gmail (in this case) doesn't recognize that the messages are all part of a single conversation, and breaks them up into multiple threads. Let's fix this minor (but rather irritating) bug. I propose that the title shown on the subject line should follow what appears when the receiving the email views that bout himself (e.g. in my case, it should show "#35023: Carlos Miranda").

**Comment by karato** [comment]
I will ask someone to take care of this task soon

**Comment by karato** [comment]
@carlosmiranda thanks for the ticket, your account was topped for 15 mins, payment 52103755

**Comment by dmzaytsev** [comment]
@karato set milestone to 3.0

**Comment by karato** [comment]
@carlosmiranda attached this issue to milestone "3.1" (let me know if this is wrong)

**Comment by karato** [comment]
> @karato set milestone to 3.0

@dmzaytsev thanks, I set it to "3.0"

**Comment by karato** [comment]
@erimerturk this task is yours, please help (see our key [principles of work](http://www.xdsd.org/2014/04/17/how-xdsd-is-different.html)). If you have any technical questions, don't hesitate to ask right here... **30 mins** is the budget of the task. This is exactly how much will be paid when the problem is [solved](http://www.xdsd.org/2014/04/15/definition-of-done.html) (no matter how much time you will actually spend). See [this](http://www.xdsd.org/2014/04/17/how-xdsd-is-different.html) for more information

**Comment by erimerturk** [comment]
@carlosmiranda i couldn't reproduce this bug. Can you help me how can i reproduce it ?

**Comment by carlosmiranda** [comment]
@erimerturk I think you'd need a second account to trigger a notification by posting on the bout

**Comment by erimerturk** [comment]
@carlosmiranda nb-friends dynamo tables have record for each bout invited friends. And when one of them change title of bout only change own bout and the others not affected this renaming. So the main problem is this i think. Should it be updated for each record or not ? It is a design decision i think.
@yegor256 what do you think, what is your suggestion ?


first update version :
![screen shot 2015-05-24 at 18 13 40](https://cloud.githubusercontent.com/assets/416283/7788391/e07a119c-0240-11e5-826f-af385bb1f308.png)
second update version :
![screen shot 2015-05-24 at 18 15 15](https://cloud.githubusercontent.com/assets/416283/7788390/e074bc10-0240-11e5-802e-b6a5d93b6a72.png)

**Comment by yegor256** [comment]
@erimerturk this is by design. I mean, the structure of DynamoDB table. each user should be able to rename his own bout and this change should not be visible to other users. But when an email is sent, we should use the title that is "owned" by the recipient of the email.

**Comment by erimerturk** [comment]
@carlosmiranda @yegor256 I need to find personal bout title from nb-friends  so i am planning to add a new method to <b>Friend </b> interface, it will find personal bout by number and alias.  it will be something like this. Is it good idea adding new method to <b>Friend</b> ?

 <code>String title(final long number) throws IOException; </code>

What do you think about this solution and what are your suggestions. ?What is the proper way to access dynamodb ? I think i  need to <b>region</b> while accessing dynamo table so what is the best way of the get it ? thank you

**Comment by yegor256** [comment]
@erimerturk I don't think you should touch/modify `Friend` interface. I think you can just "find" a friend in the `base` and then get his inbox and find bout there, and get title from it

**Comment by erimerturk** [comment]
@yegor256 i am not able to get Friend's inbox. I need to get user and then user's inbox.  <code>Base.user(URN urn)</code> is expecting <b>URN</b> to get user. But <code>Friend</code> has <code>alias</code> not <code>URN</code> . So i am not able to access user, there should be method or way getting user <b> by alias </b>. Should i add a method to <code>Base</code> interface ? any suggestion, thank you ?

**Comment by erimerturk** [comment]
@yegor256 any advice, i am blocked by <b>URN</b> problem. we need to provide <b>URN</b> for friend or find User by <b>alias</b>. What do you think ?

**Comment by yegor256** [comment]
@erimerturk I'm not sure... thinking...

**Comment by erimerturk** [comment]
@yegor256 @karato should we pause this issue until find a proper technical solution ? My concurrent task limit is just 2 and this task is holding one of them.

**Comment by yegor256** [comment]
@dmzaytsev I think we should postpone it... I don't know how to do it right, but we have to find a proper solution

**Comment by dmzaytsev** [comment]
@karato this is postponed task

**Comment by dmzaytsev** [comment]
@karato please set milestone to 3.1

**Comment by karato** [comment]
> @karato this is postponed task

@dmzaytsev OK, I labeled it as "postponed" tag

**Comment by karato** [comment]
> @karato this is postponed task

@dmzaytsev someone else will help in this task, no problem at all

**Comment by karato** [comment]
> @karato please set milestone to 3.1

@dmzaytsev milestone is set to 3.1
